tests/run-pass/atomic.rs

```diff
@@ -58,10 +71,7 @@ fn atomic_u64() {
 
     ATOMIC.store(1, SeqCst);
     assert_eq!(ATOMIC.compare_exchange(0, 0x100, AcqRel, Acquire), Err(1));
-    assert_eq!(
-        ATOMIC.compare_exchange_weak(1, 0x100, AcqRel, Acquire),
-        Ok(1)
-    );
+    compare_exchange_weak_loop!(ATOMIC, 1, 0x100, AcqRel, Acquire);
     assert_eq!(ATOMIC.load(Relaxed), 0x100);
 }
```

Comment:
Can we also have a test to ensure that it sometimes spuriously fails? E.g. we could wrap the retry loop into another loop to do 10 `compare_exchange_weak` in a row, and then check that there was at least one failure between the 10 of them.

Reply:
I'll add a test that it sometimes fails. I did 20 iterations instead of ten, since a 1 in 1024 (2^10) failure seems like it could rarely happen but 2^20 (1 in 1048576) seems sufficiently unlikely.